app/database/operator/server_data_operator/handlers/custom_profile.ts

```diff
@@ -40,6 +40,12 @@ const CustomProfileHandler = <TBase extends Constructor<ServerDataOperatorBase>>
         }
 
         const createOrUpdateRawValues = getUniqueRawsBy({raws: fields, key: 'id'});
+        try {
+            const deletedFields = fields.filter((field) => field.delete_at !== 0).map((field) => deleteCustomProfileAttributesByFieldId(this.database, field.id));
+            await Promise.all(deletedFields);
+        } catch (error) {
+            logError('Error deleting custom profile attributes by field id', error);
+        }
```

Comment:
we can keep it like this until https://github.com/mattermost/mattermost-mobile/pull/8802 gets merged

Reply:
https://mattermost.atlassian.net/browse/MM-64352